source/tools/GenContainer.rst

```diff
@@ -16,39 +16,95 @@ A container is an encapsulation of multiple components as depicted in the follow
 Image Types
 ===========
 
-Normal
-^^^^^^
+1. Normal
+^^^^^^^^^
 
-This type of container contains a single image in it. Image format can be ELF, Multiboot, Multiboot-2, UEFI-PI FV, or PE32.
+This type of container contains a single component in it. Component format can be ELF, UEFI-PI FV, or PE32.
 
-Classic
-^^^^^^^
+ - example command::
 
-This image type is used to boot a typical Linux image. This image format combines a kernel command line and the kernel image.
-Optionally more files like an initrd, ACPI tables, firmware files could be added. 
+    python GenContainer.py create -t NORMAL -cl UEFI:UefiPld.fd
+
+2. Classic Linux
+^^^^^^^^^^^^^^^^
+
+This container type is used to boot a typical Linux operating system. This container is made up of atleast two
+components - the kernel command line, and kernel image.
+
+**SBL expects the components inside the container in the following order: kernel command line, kernel image,
+and the optional initrd.**
 
-The kernel command line, kernel image, and the optional initrd are in a **fixed order**.
   * Dummy files need to be provided if one of the component is not in use. 
   * For example, if initrd is not used, a dummy file needs to be provided in place of the initrd.
 
+Optionally, more components like ACPI tables, firmware blobs, etc. can be added to the container. These blobs
+will be loaded by SBL into memory.
+
+Some device drivers may require binary blobs for initialization. These are typically loaded from the filesytem.
+In some use cases like fast boot, having access to these blobs before the filesystem is mounted may be desirable.
+Packaging blobs with the container provides these drivers a way to get the blobs without waiting for the filesystem
+to be mounted. SBL can pass the memory address of the loaded blob through the Linux command line. For this, SBL
```

Comment:
When these additional blobs are packaged with the container, SBL can load them to memory and pass the memory address through the kernel command line.